x-pack/legacy/plugins/security/index.js

```diff
@@ -33,31 +29,22 @@ import { watchStatusAndLicenseToInitialize } from '../../server/lib/watch_status
 import { SecureSavedObjectsClientWrapper } from './server/lib/saved_objects_client/secure_saved_objects_client_wrapper';
 import { deepFreeze } from './server/lib/deep_freeze';
 import { createOptionalPlugin } from '../../server/lib/optional_plugin';
+import { KibanaRequest } from '../../../../src/core/server';
 
+let defaultVars;
 export const security = (kibana) => new kibana.Plugin({
   id: 'security',
   configPrefix: 'xpack.security',
   publicDir: resolve(__dirname, 'public'),
   require: ['kibana', 'elasticsearch', 'xpack_main'],
 
   config(Joi) {
-    const providerOptionsSchema = (providerName, schema) => Joi.any()
-      .when('providers', {
-        is: Joi.array().items(Joi.string().valid(providerName).required(), Joi.string()),
-        then: schema,
-        otherwise: Joi.any().forbidden(),
-      });
-
     return Joi.object({
```

Comment:
__note:__ potentially we could move entire config schema to NP in this PR and expose values we need in LP through setup contract and remove them as soon as we fully migrate to NP, but I didn't do that since we don't have a proper config deprecations support in NP yet (required for `authorization.legacyFallback.enabled` deprecation transformation).